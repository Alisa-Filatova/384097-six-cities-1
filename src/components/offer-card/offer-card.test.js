import React from 'react';
import renderer from 'react-test-renderer';
import OfferCard from './offer-card.jsx';

const mock = {
  id: 123,
  title: `Beautiful & luxurious apartment at great location`,
  img: `img/apartment-01.jpg`,
  isPremium: true,
  price: 120,
  stars: 4,
  type: `Apartment`,
  isInBookmarks: false,
  coordinates: [52.3909553943508, 4.929309666406198],
};

describe(`OfferCard`, () => {
  it(`renders correctly`, () => {
    const onTitleClickHandler = jest.fn();
    const onImageClickHandler = jest.fn();
    const mouseOverHandler = jest.fn();
    const mouseLeaveHandler = jest.fn();

    const card = renderer.create(
        <OfferCard
          offer={mock}
          onOfferTitleClick={onTitleClickHandler}
          onOfferImgClick={onImageClickHandler}
          onMouseOver={mouseOverHandler}
          onMouseOut={mouseLeaveHandler}
        />
    ).toJSON();

    expect(card).toMatchSnapshot();
  });
});
