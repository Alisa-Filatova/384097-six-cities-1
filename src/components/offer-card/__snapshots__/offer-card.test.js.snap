// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard renders correctly 1`] = `
<article
  className="cities__place-card cities__card place-card"
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      onClick={[Function]}
    >
      <img
        alt="Beautiful & luxurious studio at great location"
        className="place-card__image"
        height={200}
        src="img/1.png"
        width={260}
      />
    </a>
  </div>
  <div
    className="cities__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <FavoriteButton
        isActive={false}
        onClick={[Function]}
      />
    </div>
    <RatingStars
      rating={4.8}
    />
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/1"
      >
        Beautiful & luxurious studio at great location
      </Link>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
